### PR TITLE
Add support for python versions 3.10 and 3.11

### PR DESCRIPTION
**Pull request by gordwinAdama**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,10 +17,10 @@ jobs:
 
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9, 3.10, 3.11]
```
Comment (pevogam) [inline .github/workflows/ci.yml:20]: Here you should use 3.10.8 or the zero will get eaten.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,10 +17,10 @@ jobs:
...
         install_variant: ["pip"]
         include:
-        - python-version: 3.10.8
+        - python-version: 3.11
```
Comment (pevogam) [inline .github/workflows/ci.yml:23]: Why this change? I would say to just add supported versions the above list should be enough. The following is just additional included variants.

**Review by pevogam — CHANGES_REQUESTED** [review]
Hi @gordwinAdama, thanks for your pull request! I left some comments below.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (gordwinAdama) [reply .github/workflows/ci.yml:20]: Ok, Thanks for the info.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (gordwinAdama) [reply .github/workflows/ci.yml:23]: Ok, So additions should be just the supported versions. Thanks.

Comment (pevogam) [reply .github/workflows/ci.yml:23]: The mix
```
          python-version: 3.10.8
          install_variant: "rpm.fedora.34"
```
must remain the same so you should revert your change here. The reason is that this defines a "test variant" that only works on Fedora 34 OS which always has exactly python 3.10.8 installed there.

File: packaging/packager_pip.sh [hunk 3]
```diff
@@ -15,6 +15,8 @@ if [[ $python_version == '3.9' ]]; then export DISABLE_AUTOPY=1; fi
 
 # environment dependencies not provided by pip
 # python3
+if [[ $python_version == '3.11' ]]; then dnf -y install python3.11 python39-devel; fi
+if [[ $python_version == '3.10.8' ]]; then dnf -y install python3.10.8 python39-devel; fi
```
Comment (pevogam) [inline packaging/packager_pip.sh:19]: This does not follow the logic of the lines below, the versions see right below.

File: packaging/packager_rpm.sh [hunk 4]
```diff
@@ -20,7 +20,7 @@ dnf -y install tesseract tesseract-devel
 dnf -y install gcc-c++
 pip3 install pytesseract==0.3.4 tesserocr==2.5.1
 # deep learning
-pip3 install torch==1.8.1 torchvision==0.9.1
+pip3 install torch==1.13.0 torchvision==0.14.0
```
Comment (pevogam) [inline packaging/packager_rpm.sh:23]: I am confused - some torch versions are 1.13.0 and others are 1.13.1.

Comment (gordwinAdama) [reply packaging/packager_rpm.sh:23]: Corrected the versions.

File: packaging/packager_pip.sh (hunk 3 above)
Comment (gordwinAdama) [reply packaging/packager_pip.sh:19]: Okay, onto it

File: packaging/packager_rpm.sh [hunk 5]
```diff
@@ -20,7 +20,7 @@ dnf -y install tesseract tesseract-devel
 dnf -y install gcc-c++
 pip3 install pytesseract==0.3.4 tesserocr==2.5.1
 # deep learning
-pip3 install torch==1.8.1 torchvision==0.9.1
+pip3 install torch==2.0.0 torchvision==0.15.1
```
Comment (pevogam) [inline packaging/packager_rpm.sh:23]: I think this is going too far in the version and it might break our code but of course would be great if it works or needs just a few adjustments.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pevogam) [reply .github/workflows/ci.yml]: This is now correct, resolving this comment.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (pevogam) [reply .github/workflows/ci.yml]: Ok, it seems you also got this right now, resolving.

File: packaging/packager_pip.sh (hunk 3 above)
Comment (pevogam) [reply packaging/packager_pip.sh]: Ok removed so resolving

File: packaging/packager_pip.sh [hunk 6]
```diff
@@ -12,9 +12,11 @@ readonly release_tag="${RELEASE_TAG:-}"
 
 # disable tests and support of backends based on python versions
 if [[ $python_version == '3.9' ]]; then export DISABLE_AUTOPY=1; fi
-
```
Comment (pevogam) [inline packaging/packager_pip.sh:15]: Please keep this empty line to separate the upper from the lower block of lines.

File: packaging/packager_pip.sh [hunk 7]
```diff
@@ -12,9 +12,11 @@ readonly release_tag="${RELEASE_TAG:-}"
...
+if [[ $python_version == '3.10.8' ]]; then export DISABLE_AUTOPY=1; fi
+if [[ $python_version == '3.11' ]]; then export DISABLE_AUTOPY=1; fi
 # environment dependencies not provided by pip
 # python3
+
```
Comment (pevogam) [inline packaging/packager_pip.sh:19]: This empty line should not have been added.

File: packaging/pip_requirements.txt [hunk 8]
```diff
@@ -1,15 +1,16 @@
 # minimal
-Pillow==8.2.0; python_version == '3.6'
 Pillow==9.0.0; python_version >= '3.7'
+torch==1.13.0; python_version == '3.7'
+torchvision==0.14.0; python_version == '3.7'
```
Comment (pevogam) [inline packaging/pip_requirements.txt:4]: The problem here is that for python_version == '3.7' this condition will be met (and these versions installed) but the below conditions will also be met and torch==2.0.1 installed too, resulting in conflict. What you have to do now is exclude this case from the case below by changing the lower line as follows
```diff
- torch==2.0.1; 'generic' not in platform_release and platform_python_implementation != "PyPy"
+ torch==2.0.1; python_version != '3.7' and 'generic' not in platform_release and platform_python_implementation != "PyPy"
```

**Review by pevogam — CHANGES_REQUESTED** [review]
I left some more comments regarding the most recent changes which are now going in the right direction.

**Comment by pevogam** [comment]
Closing this via #52 which builds on top.
